example/advection/advection_driver.cpp

```diff
@@ -44,106 +47,155 @@ AdvectionDriver::AdvectionDriver(ParameterInput *pin, ApplicationInput *app_in,
-  auto &sc1 = pmb->real_containers.Get(stage_name[stage]);
-
-  auto start_recv = tl.AddTask(none, &Container<Real>::StartReceiving, sc1.get(),
-                               BoundaryCommSubset::all);
-
-  auto advect_flux = tl.AddTask(none, advection_package::CalculateFluxes, sc0);
-
-  auto send_flux =
-      tl.AddTask(advect_flux, &Container<Real>::SendFluxCorrection, sc0.get());
-  auto recv_flux =
-      tl.AddTask(advect_flux, &Container<Real>::ReceiveFluxCorrection, sc0.get());
-
-  // compute the divergence of fluxes of conserved variables
-  auto flux_div = tl.AddTask(recv_flux, parthenon::Update::FluxDivergence, sc0, dudt);
-
-  // apply du/dt to all independent fields in the container
-  auto update_container =
-      tl.AddTask(flux_div, UpdateContainer, pmb, stage, stage_name, integrator);
-
-  // update ghost cells
-  auto send =
-      tl.AddTask(update_container, &Container<Real>::SendBoundaryBuffers, sc1.get());
-  auto recv = tl.AddTask(send, &Container<Real>::ReceiveBoundaryBuffers, sc1.get());
-  auto fill_from_bufs = tl.AddTask(recv, &Container<Real>::SetBoundaries, sc1.get());
-  auto clear_comm_flags = tl.AddTask(fill_from_bufs, &Container<Real>::ClearBoundary,
-                                     sc1.get(), BoundaryCommSubset::all);
-
-  auto prolong_bound = tl.AddTask(
-      fill_from_bufs,
-      [](MeshBlock *pmb) {
-        pmb->pbval->ProlongateBoundaries(0.0, 0.0);
-        return TaskStatus::complete;
-      },
-      pmb);
-
-  // set physical boundaries
-  auto set_bc = tl.AddTask(prolong_bound, parthenon::ApplyBoundaryConditions, sc1);
-
-  // fill in derived fields
-  auto fill_derived =
-      tl.AddTask(set_bc, parthenon::FillDerivedVariables::FillDerived, sc1);
-
-  // estimate next time step
-  if (stage == integrator->nstages) {
-    auto new_dt = tl.AddTask(
-        fill_derived,
-        [](std::shared_ptr<Container<Real>> &rc) {
-          auto pmb = rc->GetBlockPointer();
-          pmb->SetBlockTimestep(parthenon::Update::EstimateTimestep(rc));
+  // note that task within this region that contains one tasklist per pack
+  // could still be executed in parallel
+  TaskRegion &single_tasklist_per_pack_region = tc.AddRegion(partitions.size());
+  for (int i = 0; i < partitions.size(); i++) {
+    auto &tl = single_tasklist_per_pack_region[i];
+    auto &mbase = pmesh->mesh_data.Get("base" + std::to_string(i));
+    auto &mc0 = pmesh->mesh_data.Get(stage_name[stage - 1] + std::to_string(i));
+    auto &mc1 = pmesh->mesh_data.Get(stage_name[stage] + std::to_string(i));
+    auto &mdudt = pmesh->mesh_data.Get("dUdt" + std::to_string(i));
+
+    // compute the divergence of fluxes of conserved variables
+    auto flux_div = tl.AddTask(none, parthenon::Update::FluxDivergenceMesh, mc0, mdudt);
+
+    // apply du/dt to all independent fields in the container
+    auto update =
+        tl.AddTask(flux_div, UpdateMeshData, stage, integrator, mc0, mbase, mdudt, mc1);
+  }
```

Comment:
While going further through the code, I'm now wondering if it would make sense to change the interface to Get/Set "Data" should include an integer (particularly the the blocks/partitions).
For the latter the use case I see will almost entirely work through indices so we could skip the step of creating label based on indices.
What do people think?

Reply:
I'm not quite sure I understand. What would the new calls look like?

Reply:
Currently strings are used in the example code as primary means to identify variable packs and encode lots of information and I'm wondering how robust this suggested use case (as we provide it in the example).
For example, the `stage_name`s (except for the `base`) are also `int` as are the partitions.
So what we have definitely works, but I see ways manually encoding information a string could break (or lead to unintended behavior).
I think what would be nice is sth. like a simple `pmesh->mesh_data.Get(stage_str, partition_int)` where the partitions are setup transparently without registering them (however, that may require more refactoring as these regions may run in parallel).

Reply:
At the expense of having a Mesh pointer in the DataCollection this now kind of works, see https://github.com/lanl/parthenon/compare/71b46aa...51e2b8e including updated doc
I'm happy to hear what people think about it. My main goal was to hide the complexity of creating/duplicating the information that is effectively already provided by creating `MeshBlockData`

Reply:
hmm interesting. Yeah, I think I like the idea of splitting it like that.